internal/kafka/internal/config/dataplane_cluster_config_test.go

```diff
@@ -521,11 +689,268 @@ func TestValidateClusterIsInKubeContext(t *testing.T) {
+				},
+			},
+			args: args{
+				clusterId: "test02",
+			},
+			want:         false,
+			wantInstType: "",
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			conf := NewClusterConfig(tt.fields.ClusterList)
+			instType, found := conf.GetClusterSupportedInstanceType(tt.args.clusterId)
+			Expect(instType).To(Equal(tt.wantInstType))
+			Expect(found).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_GetManualClusters(t *testing.T) {
+	type fields struct {
+		ClusterList ClusterList
+	}
+	type args struct {
+		clusterId string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   []ManualCluster
+	}{
+		{
+			name: "should return manual clusters slice from config",
+			fields: fields{
+				ClusterList: ClusterList{
+					ManualCluster{ClusterId: "test01", Schedulable: true},
+				},
+			},
+			args: args{
+				clusterId: "test01",
+			},
+			want: ClusterList{
+				ManualCluster{ClusterId: "test01", Schedulable: true},
+			},
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			conf := NewClusterConfig(tt.fields.ClusterList)
+			Expect(conf.GetManualClusters()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_IsReadyDataPlaneClustersReconcileEnabled(t *testing.T) {
+	tests := []struct {
+		name                                  string
+		enableReadyDataPlaneClustersReconcile bool
+		want                                  bool
+	}{
+		{
+			name:                                  "should return true if EnableReadyDataPlaneClustersReconcile is set to true",
+			enableReadyDataPlaneClustersReconcile: true,
+			want:                                  true,
+		},
+		{
+			name:                                  "should return false if EnableReadyDataPlaneClustersReconcile is set to false",
+			enableReadyDataPlaneClustersReconcile: false,
+			want:                                  false,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			conf := NewDataplaneClusterConfig()
+			conf.EnableReadyDataPlaneClustersReconcile = tt.enableReadyDataPlaneClustersReconcile
+			Expect(conf.IsReadyDataPlaneClustersReconcileEnabled()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_ReadFiles(t *testing.T) {
+	type fields struct {
+		config *DataplaneClusterConfig
+	}
+
+	tests := []struct {
+		name     string
+		fields   fields
+		modifyFn func(config *DataplaneClusterConfig)
+		wantErr  bool
+	}{
+		{
+			name: "should return an error with misconfigured ImagePullDockerConfig",
+			fields: fields{
+				config: NewDataplaneClusterConfig(),
+			},
+			modifyFn: func(config *DataplaneClusterConfig) {
+				config.ImagePullDockerConfigFile = "invalid"
+				config.ImagePullDockerConfigContent = ""
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with EnableReadyDataPlaneClustersReconcile set to true and invalid DataPlaneClusterConfigFile",
```

Comment:
Reading this test case and the ones below it, I've an impression that `EnableReadyDataPlaneClustersReconcile` controls how files are read but this is not actually the case.

Reply:
The `EnableReadyDataPlaneClustersReconcile` is independent on how files are read. I'd suggest we drop the flag from the test case name and from setting it up in the test.

Reply:
what's tested here depends on whether the flag is set or not, e.g. if it's set, `DataPlaneClusterConfigFile` will be read (otherwise it won't), so I think the test is valid

Reply:
> what's tested here depends on whether the flag is set or not, e.g. if it's set, `DataPlaneClusterConfigFile` will be read (otherwise it won't), so I think the test is valid

Can you point where the `EnableReadyDataPlaneClustersReconcile` is used for what you described above? I couls be missing something obvious but I did a quick check and I couldn't see the flag being used for that.

Reply:
just looked again at the code and I see your point about dropping the `EnableReadyDataPlaneClustersReconcile` now. Sorry for confusion. I am removing it and running the tests again

Reply:
Thanks!